source/asyncify.d.ts

```diff
@@ -0,0 +1,29 @@
+import {PromiseValue} from './promise-value';
+import {SetReturnType} from './set-return-type';
+
+/**
+Create an async version of the given function type.
+
```

Comment:
```suggestion

Use-case: you have two functions, one synchronous and one asynchronous, that do the same thing. Instead of having to duplicate the type specification, you can use `Asyncify` to reuse the synchronous type.
```